pkg/api/v1/types.go

```diff
@@ -532,12 +532,21 @@ const (
 	ClaimLost PersistentVolumeClaimPhase = "Lost"
 )
 
+type PropagationMode string
+
+const (
+	PropagationPrivate PropagationMode = "private"
+	PropagationShared  PropagationMode = "shared"
+)
+
 // Represents a host path mapped into a pod.
 // Host path volumes do not support ownership management or SELinux relabeling.
 type HostPathVolumeSource struct {
 	// Path of the directory on the host.
 	// More info: http://releases.k8s.io/HEAD/docs/user-guide/volumes.md#hostpath
 	Path string `json:"path" protobuf:"bytes,1,opt,name=path"`
+	// Mount the host path with propagation mode specified. Docker only.
```

Comment:
Strike the Docker only.  We need to know that it is viable on at least other container-based runtimes - rkt and OCI.  What are the API equivalents in those?

@yujuhong this might be another feature flag for CRI :-/

Abstractly, this is "should mountpoints in this volume be exposed to the host or private".  Can we look at the naming and semantic through that lens?  What if this was "PublishMounts bool" or something?  Maybe even a flag that defaults to true like "PrivateMounts".  What naming *explains* this best?